src/client/containers/ProductPage/ProductPage.Container.js

```diff
@@ -1,12 +1,79 @@
+/* eslint-disable no-console */
+/* eslint-disable radix */
 import React from 'react';
+import { useParams } from 'react-router-dom';
+import { ProductDetails } from '../../components/ProductDetails/ProductDetails.component';
+import Carousel from '../../components/Carousel/Carousel.component';
+import ButtonComponent from '../../components/Button/Button.component';
 import './ProductPage.Style.css';
 
 const ProductPageContainer = () => {
+  const { id } = useParams();
+  const [product, setProduct] = React.useState({});
+  const [similarProduct, setSimilarProduct] = React.useState([]);
+
+  React.useEffect(() => {
+    fetch(`api/products/${id}`)
+      .then((res) => res.json())
+      .then((data) => {
+        setProduct(data[0]);
+      })
+      .catch((e) => {
+        console.log(e);
+      });
+  }, [id]);
+
+  React.useEffect(() => {
+    fetch(`api/categories/${product.category_id}`)
```

Comment:
Use the custom hook: `useFetchApi`.

Reply:
I have made changes in first usecase above, but in this case, this hook does not fit my scenario. 
would you be kind to explain to how use useFetchApi hook in this case. 

because I need to get first "category_id" from first fetch, then in second fetch, i need to get "category name" and in third fetch i need to get similar products.. 
so i dont know how to use this hook for my usecase " to get similar product"

Reply:
So As discussed you can add optional dependency in custom hook useFEtchApi. In your case 1st fetch needs id ,2nd - product data, and 3rd - category name

Reply:
Hi Julia, I tried to implement with your suggestion and not able to make hook. so I am keeping the working code.

Reply:
done with useFetchAp